sklearn/impute/_base.py

```diff
@@ -205,7 +206,8 @@ class SimpleImputer(_BaseImputer):
     upon :meth:`transform` if strategy is not "constant".
 
     """
-    def __init__(self, missing_values=np.nan, strategy="mean",
+    @_deprecate_positional_args
+    def __init__(self, missing_values=np.nan, *, strategy="mean",
```

Comment:
I'd like the first arg here to be keyword only. That would allow us to consider making strategy (or n_neighbours for KNNImputer) positional.